h3i/src/frame.rs

```diff
@@ -485,25 +555,118 @@ impl ExpectedFrame {
     }
 }
 
+impl Debug for ExpectedFrame {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let repr = match &self.comparator {
+            Comparator::Frame(frame) => format!("{frame:?}"),
+            Comparator::Fn(_) => "closure".to_string(),
+        };
+
+        write!(
+            f,
+            "{}",
+            format!(
+                "ExpectedFrame {{ stream_id: {}, comparator: {repr} }}",
+                self.stream_id
+            )
+        )
+    }
+}
+
+impl PartialEq for ExpectedFrame {
+    fn eq(&self, other: &Self) -> bool {
+        match (&self.comparator, &other.comparator) {
+            (Comparator::Frame(this_frame), Comparator::Frame(other_frame)) =>
+                self.stream_id == other.stream_id && this_frame == other_frame,
+            _ => false,
+        }
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
+    use quiche::h3::frame::Frame;
 
     #[test]
-    fn test_equivalence() {
-        let this = ExpectedFrame::new(0, vec![
+    fn test_header_equivalence() {
+        let mut this = ExpectedFrame::new(0, vec![
             Header::new(b"hello", b"world"),
             Header::new(b"go", b"jets"),
         ]);
-        let other = ExpectedFrame::new(0, vec![
+        let mut other: H3iFrame = vec![
             Header::new(b"hello", b"world"),
             Header::new(b"go", b"jets"),
             Header::new(b"go", b"devils"),
-        ]);
+        ]
+        .into();
+        assert!(this.is_equivalent(&other));
+
+        this = ExpectedFrame::new(0, other);
+        other =
+            vec![Header::new(b"hello", b"world"), Header::new(b"go", b"jets")]
+                .into();
+        // `other` does not contain the `go: devils` header, so it's not
+        // equivalent to `this.
+        assert!(!this.is_equivalent(&other));
+    }
```

Comment:
I'm finding it hard to follow this example. Can we just split into 2 tests or does the sequencing matter?

Reply:
Split it into 2 tests